contracts/RelayHub.sol

```diff
@@ -247,6 +250,26 @@ contract RelayHub is RelayHubApi {
         require( decoded_tx1.nonce == decoded_tx2.nonce, "Different nonce");
         require(addr1 == addr2, "Different signer");
         require(keccak256(abi.encodePacked(decoded_tx1.data)) != keccak256(abi.encodePacked(decoded_tx2.data)), "tx.data is equal" ) ;
+        penalize_internal(addr1);
+    }
+
+    function penalize_illegal_transaction(bytes memory unsigned_tx1, bytes memory sig1) public {
+        // Externally-owned accounts that are registered as relays are not allowed to perform
+        // any transactions other than 'relay' and 'register_relay'. They have no legitimate
+        // reasons to do that, so this behaviour is too suspicious to be left unattended.
+        // It is enforced by penalizing the relay for a transaction that we consider illegal.
+
+        Transaction memory decoded_tx1 = decode_transaction(unsigned_tx1);
+        if (decoded_tx1.to == address(this)){
+            bytes4 selector = GsnUtils.getMethodSig(decoded_tx1.data);
+            require (selector != this.relay.selector && selector != this.register_relay.selector, "Legal relay transaction");
```

Comment:
Please add a comment explaining that if we add another valid function call to RelayHub, we must add its selector here, otherwise the relay is penalizable...